lib/Rest/HttpRequest.cpp

```diff
@@ -895,7 +898,9 @@ VPackSlice HttpRequest::payload(VPackOptions const* options) {
     validationOptions.disallowExternals = true;
     validationOptions.disallowCustom = true;
     VPackValidator validator(&validationOptions);
-    validator.validate(_payload.data(), _payload.length()); // throws on error
+    if(!_validatedPayload) {
```

Comment:
The entire assembling of the `validationOptions` and the creation of the `validator` object could also go into this `if` clause, because these objects are only used inside the `if`.
```suggestion
    if (!_validatedPayload) {
```